api/routes/user.js

```diff
@@ -0,0 +1,127 @@
+const router = require('express').Router()
+const User = require('../models/user')
+const bcrypt = require('bcrypt')
+const jwt = require('jsonwebtoken')
+const { SECRET_KEY } = process.env
+
+router.post('/signup', async (req, res, next) => {
+    const status = 201
+    const badRequest = 400
+    let userExists = false
+    try{
+        const{user, password, admin} = req.body
+        if(!user){
+            throw new Error('A user name is required')
+        }
+        if(password.length < 8){
+            throw new Error('Passwords must be at least 8 characters')
+        }
+        const existingUser = await User.findOne({user})
+        if(existingUser){
+            userExists = true
+            throw new Error('That username is already taken.  Please try another.')
+        }
+        
+        const salt = 10
+        const bcryptPassword = await bcrypt.hash(password, salt)
+        const response = await User.create({
+            user:user, 
+            password:bcryptPassword,
+            admin: admin
```

Comment:
You typically won't want to allow for a user to be able set _themselves_ as an admin. For future applications, I would only allow admin via updating a user.